### PR TITLE
fix: sort Gantt timeline tasks by start time

### PR DESCRIPTION
**Pull request by tacascer**
## Summary
- Sort timeline tasks by `startTimestamp` so the Gantt chart displays tasks in chronological order (earliest at top)
- Applies to both the normal and zero-duration edge cases in `getTimeline()`

## Test plan
- [x] Unit tests pass (`aspect test //angular/projects/build-scan-web:test`)
- [x] Visual verification: open a build scan detail page and confirm tasks are ordered top-to-bottom by start time

### CODE REVIEW

File: angular/projects/build-scan-web/src/app/scans/scan-detail.component.ts [hunk 1]
```diff
@@ -300,9 +303,13 @@ export class ScanDetailComponent {
       };
     }
 
+    const sorted = [...tasks].sort(
+      (a: any, b: any) => a.node.startTimestamp - b.node.startTimestamp
+    );
```
[the remark this answers is not in the document]
Comment (tacascer) [reply angular/projects/build-scan-web/src/app/scans/scan-detail.component.ts]: Added a test that provides tasks in reverse order (`:late`, `:early`, `:middle`) and asserts they render sorted by startTimestamp. Fixed in 65d11a1.
